### DISCUSSION TITLE
Umi & Bigfish 想法征集 05

### DISCUSSION THREAD

**Discussion by sorrycc** [umijs/umi#10750] [Ideas]
> Why this issue? 计划在 Umi Contributor 群增加一条规则。每人每周贡献 2 个改进 Umi 或 Bigfish 的想法（辟起、辟殊、云谦），外包同学减半（1 个），外部 Contributor 随意，我每周三会在群里发个 issue 做收集。

直接回复即可。（含内部敏感信息的可以私发我）

**Comment by xierenyuan** [comment]
配置全部迁移 zod 后 如果对 zod 做一下描述的扩展, 生成对应的 md 文件, 是否可用于替换现在文档的  配置页  类似 nuxt 框架现在的做法, 好处是配置的变化不用在关注 文档变更

**Reply by fz6m** [reply]
目前深层的 schema 结构里 desc 还不支持（zod to ts，用 ts 还是拼字符串目前都不支持），需要继续探究，我还没深入。

**Comment by sorrycc** [comment]
- Bigfish 新增 scripts，`tnpm run local:link path/to/project` 用于调试本地项目
- 新增 Bigfish create 命令，不依赖 basement-cli 创建本地项目（用于验证）
- 页面组件支持 default export 和 names exports（Page）？
- 新增 OpenAI 配置项，把 ChatGPT 带到框架里
	- I want you to act as a code improver. Base on a ${language} error message and error stack, send me resolution sample code.
	- [error-stack-gpt/explain.ts at main · RaoHai/error-stack-gpt · GitHub](https://github.com/RaoHai/error-stack-gpt/blob/main/pages/api/explain.ts#L21)

**Comment by xyuanbuilds** [comment]
恢复umi clear命令，用于清空所有临时文件、缓存等，快速恢复到install后的初始状态

**Reply by sorrycc** [reply]
👍🏻，记录了。

**Reply by PeachScript** [reply]
这个感觉可以和 umi cache 命令合并？有 PR 了：https://github.com/umijs/umi/pull/10730

**Comment by PeachScript** [comment]
1. `legacy` 配置项可以集成 antd v5 的旧版浏览器兼容能力以降低用户成本？需要额外注入 runtime plugin：https://ant.design/docs/react/compatible-style-cn#css-%E9%80%BB%E8%BE%91%E5%B1%9E%E6%80%A7
2. Umi CI 增加 `father doctor` 校验

**Reply by fz6m** [reply]
`1` 感觉可以做，弄到 antd 插件里，检测 `legacy` 选项是否开启。

**Reply by sorrycc** [reply]
1 记录了
2 我记录到新手任务里

**Comment by stormslowly** [comment]
* 脑暴：bigfish 内部发版，可以通过 tag 的方式回滚，那我们能不能做到以灰度的方式发布。
* 如何在 Node 端计算 HMR 的时间，并Telemetry 上报。

**Reply by xierenyuan** [reply]
HMR 时间 困扰好久呢 +1
